1_fetch.smk

```diff
@@ -1,43 +1,59 @@
 import os
 
-configfile: "1_fetch/fetch_config.yaml"
 
-
-def get_mntoha_input_files(wildcards):
-    '''List all MNTOHA files to download
-    Includes metadata, temperature observations, and drivers
-
-    :param wildcards: Snakemake wildcards (unused)
-    :returns: list of filepaths to be downloaded
-
-    '''
-    out_dir = "1_fetch/out"
-    categories = config["mntoha"].keys()
-    input_files = []
-    # include files under files key in each category
-    for category in categories:
-        for filename in config["mntoha"][category]["files"]:
-            input_files.append(os.path.join(out_dir, category + "_mntoha", filename))
-    # include driver files: 3 types x 13 suffixes = 39 files
-    for driver_type in config["mntoha"]["drivers"]["driver_types"]:
-        for suffix in config["mntoha"]["drivers"]["suffixes"]:
-            input_files.append(os.path.join(out_dir, "drivers_mntoha", f"{driver_type}_{suffix}.zip"))
-    return input_files
+# Download lake_metadata.csv
+# This is a checkpoint because lake_metadata.csv is needed to determine 
+# later outputs; namely, the lake_sequence file names.
+checkpoint fetch_mntoha_metadata:
+    input:
+        "1_fetch/in/pull_date.txt",
+    output:
+        "1_fetch/out/lake_metadata.csv"
+    params:
+        sb_id = "5e5c1c1ce4b01d50924f27e7"
+    script:
+        "1_fetch/src/sb_fetch.py"
 
 
-rule fetch_all:
+# Download MNTOHA temperature observations zip file
+rule fetch_mntoha_obs:
     input:
-        "1_fetch/in/pull_date.txt",
-        get_mntoha_input_files
+        "1_fetch/in/pull_date.txt"
+    output:
+        "1_fetch/tmp/obs_mntoha/temperature_observations.zip"
+    params:
+        sb_id = "5e5d0b68e4b01d50924f2b32"
+    script:
+        "1_fetch/src/sb_fetch.py"
 
 
-rule fetch_mntoha_data_file:
+# Download MNTOHA meteorological drivers, clarity, and ice flag zip files
+rule fetch_mntoha_dynamic:
     input:
         "1_fetch/in/pull_date.txt"
     output:
-        "1_fetch/out/{file_category}_mntoha/{file}"
+        "1_fetch/tmp/dynamic_mntoha/{file}"
     params:
-        sb_id = lambda wildcards: config["mntoha"][wildcards.file_category]["sb_id"]
+        sb_id = "5e5d0b96e4b01d50924f2b34"
```

Comment:
okay, so we're now just automatically downloading everything from this `sb_id`, right?

Reply:
Not everything - only the file requested.

Reply:
Oh okay -- where is the file requested now that there is no `configfile`?

Reply:
The file is inferred from the output. So, if I run the command
```
snakemake -c1 1_fetch/tmp/dynamic_mntoha/ice_flags_01_N48.00-49.50_W89.50-97.25.zip
```
then only that single zip file is downloaded from ScienceBase. 

1. Snakemake matches the filename to the pattern in `output` of the rule `fetch_mntoha_dynamic`.
2. Snakemake runs the script `sb_fetch.py`
3. That script calls `main` with two arguments: (1) the ScienceBase ID taken from `params` of the rule, and (2) the zip file name taken from `output` of the rule.

Reply:
Ahh I see now. I was thinking that, as in the last PR, all files would be downloaded in order to 'complete' the rules outlined in `1_fetch.smk`, but now I see that for these files, as with the temp obs, snakemake is not actually downloading the files until they are needed by rules in `2_process.smk`. So essentially `1_fetch.smk` provides the 'recipe' for how to get the files, but doesn't actually download and unzip files until a rule in `2_process.smk` has input that then triggers one of the `1_fetch.smk` 'recipes' -- does that sound right?

Reply:
In the full pipeline, though, `unzip_archive` ends up calling `fetch_mntoha_dynamic` and providing the file name. The DAG above shows that relationship. `mntoha_lake_sequences` depends on `unzip_archive`, and `mntoha_lake_sequences`'s input function `dynamic_filenames` is truly where the file name is built.

Reply:
Yes, what you said is right!